multimodal/src/autogluon/multimodal/utils/model.py

```diff
@@ -139,6 +139,13 @@ def select_model(
     for model_name in selected_model_names:
         logger.debug(f"model dtypes: {getattr(config.model, model_name).data_types}")
 
+    # clean up unused model configs
+    model_keys = list(config.model.keys())
+    model_keys.remove("names")
```

Comment:
Should we still keep the `names` for clarity?

Reply:
`names` is necessary for customizing the models to use. For example, https://github.com/autogluon/autogluon/blob/master/multimodal/src/autogluon/multimodal/presets.py#L152

Reply:
Yeah, I misread the logic.

Is the code 
```python
model_keys = list(config.model.keys())
model_keys.remove("names")
for model_name in model_keys:
    if model_name not in selected_model_names:
        delattr(config.model, model_name)
```

same as the following?

```python
for model_name in model_keys:
    if model_name not in selected_model_names + ["names"]:
        delattr(config.model, model_name)


```

Also, a side note is that I thought it may be unsafe to use `delattr` for a DictConfig object. But I checked the test-case: https://github.com/omry/omegaconf/blob/d02a0c10cbba8b01c92e1faffa995ebf2da4abe2/tests/test_basic_ops_dict.py#L88, it's a valid operator for DictConfig.

Reply:
If we don't remove key `names`, we need to add the if-pass logic inside the following for loop. They basically have the same effect.

Reply:
For me, I just got confused when seeing `.remove("name")` as the purpose of the function is to clean up the keys.

Reply:
> 

I see. `model_keys` is a temporary variable for one-time usage and will not affect other logics.

Reply:
Updated by putting `names` inside the for-loop.